### PR TITLE
[SM-537] add local storage persistence for onboarding tasks

### PR DESCRIPTION
**Pull request by willmartian**
## Type of change

```
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Undoing a task item (e.g. deleting a secret) shouldn't update the onboarding UI. Added a local storage item to cache completed tasks. 

## Code changes

- **bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.component.ts:** Add function that retrieves and updates tasks with state service
- **libs/common/src/services/state.service.ts:** Update state service with new methods for onboarding tasks
- **libs/common/src/models/domain/account.ts:** Add tasks to account model

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Comment by willmartian** [comment]
I wasn't sure if the [new state service methods](https://github.com/bitwarden/clients/pull/4880/files#diff-6bc5c3812ec34b35fb75ae133d450d4e6fc7bc8fcf9116076682104eb60a1fc1R361-R367) should import the [Task type ](https://github.com/bitwarden/clients/pull/4880/files#diff-5e0f423b8b42f70744b52f49e22dfd09d1484e6505fddadcdf0bbe0a50ae7608R36-R38) from the overview component or just use a generic `Record`. 

I went with record because the state service isn't currently importing from outside of `/libs` and it felt weird to move the type definition outside of SM.

**Review by Hinton — DISMISSED** [review]

File: bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.component.ts [hunk 1]
```diff
@@ -90,10 +87,7 @@ export class OverviewComponent implements OnInit, OnDestroy {
   }
 
   ngOnInit() {
-    const orgId$ = this.route.params.pipe(
-      map((p) => p.organizationId),
-      distinct()
-    );
+    const orgId$ = this.route.params.pipe(map((p) => p.organizationId));
```
Comment (Hinton) [inline bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.component.ts:90]: We should use `distinctUntilChanged`.

Comment (willmartian) [reply bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.component.ts]: I think it was meant to be `distinctUntilChanged` to begin with. Fixed.

**Review by Hinton — APPROVED** [review]

File: bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.component.html [hunk 2]
```diff
@@ -2,72 +2,78 @@
...
 
-<div *ngIf="view$ | async as view; else spinner">
-  <sm-onboarding [title]="'getStarted' | i18n" *ngIf="showOnboarding" (dismiss)="hideOnboarding()">
-    <sm-onboarding-task
-      [title]="'createServiceAccount' | i18n"
-      (click)="openServiceAccountDialog()"
-      icon="bwi-cli"
-      [completed]="view.tasks.createServiceAccount"
+<ng-container *ngIf="!loading; else spinner">
+  <div *ngIf="view$ | async as view; else spinner">
```
Comment (Hinton) [inline bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.component.html:6]: Can we just combine this into one if statement?

`<div *ngIf="!loading && view$ | async as view; else spinner">`

**Review by Hinton — APPROVED** [review]

**Pull request merged**